stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -300,6 +300,23 @@ func.func @triangular_solve(%arg0: tensor<10x5x4x4xf32>, %arg1: tensor<10x5x4x4x
   func.return %1 : tensor<10x5x4x4xindex>
 }
 
+// -----
+
+// CHECK-LABEL: func @dot_general
+func.func @dot_general(%arg0: tensor<2x3x4xf32>, %arg1: tensor<2x3x5xf32>) -> tensor<2x4x5xindex> {
```

Comment:
You may test this one separately by:
1. Make new file `/tmp/local_test.mlir` with only this test inside it.
2. after build successfully, run it by
```
test_mlir=/tmp/local_test.mlir

./build/bin/stablehlo-opt $test_mlir \
-verify-diagnostics -split-input-file -allow-unregistered-dialect
```
It will output the real output so that you can check that with your `// CHECK: ...` commends.

dont's forget to add `// -----` before and after each test to split with is necessary. 
After you find your `FileCheck`, the full test command will be:
```
test_mlir=/tmp/local_test.mlir

./build/bin/stablehlo-opt $test_mlir \
-verify-diagnostics -split-input-file -allow-unregistered-dialect \
| FileCheck $test_mlir
```
Feel free to have a gvc with me and debug this together.